Operation V2 Migration Guide Updates (#29084)
DataDog/documentation: content/en/opentelemetry/guide/migrate/migrate_operation_names.md

Datadog has introduced new logic for generating operation names for OpenTelemetry traces, controlled by the `enable_operation_and_resource_name_logic_v2` feature flag. This new logic improves trace visibility in service pages and standardizes operation naming according to the rules outlined below.

<div class="alert alert-danger">
<strong>Breaking Change:</strong> When this new logic is active (either by opting-in or future default), it is a breaking change for monitors or dashboards that reference operation names based on the old conventions. You must update your monitors and dashboards to use the new naming conventions described in <a href="#new-mapping-logic">New mapping logic</a>. If you cannot update them yet, you can <a href="#disabling-the-new-logic-opt-out">opt out</a> .
</div>

## Default rollout schedule

The `enable_operation_and_resource_name_logic_v2` feature flag controls this new logic. It is enabled by default starting in the following versions:

- **Datadog Distribution of OpenTelemetry (DDOT) Collector**: Datadog Agent v7.65+
- **OpenTelemetry Collector**: OTel Collector v0.126.0+
- **OTel to Datadog Agent (OTLP)**: Datadog Agent v7.66+

> Prior to these releases, you must explicitly [enable the logic](#enabling-the-new-logic-opt-in).

Reviewer: ```suggestion
If you are using an earlier version and want to use the new logic without upgrading, you can [explicitly enable the new logic](#enabling-the-new-logic-opt-in).
```

As-is, this sentence means "Prior to these releases, you must explicitly enable the logic, in all circumstances" but I think that you actually mean "Prior to these releases, if you want to use the new logic, you must explicitly enable the new logic."

## New mapping logic

When the `enable_operation_and_resource_name_logic_v2` flag is active, the following table shows how operation names are determined based on span attributes and kind. The system processes conditions from top to bottom and uses the first matching rule.

For example, with the new logic active, a span previously named `go.opentelemetry.io_contrib_instrumentation_net_http_otelhttp.server` is now named `http.server.request`.

> | Conditions on Span Attributes                         | Span Kind                       | Resulting Operation Name      |

Reviewer: ```suggestion
| Condition on Span Attributes                         | Span Kind                       | Resulting Operation Name      |
```

      "old_name1": "new_name"
{{< /highlight >}}

2. Remove these environment variables if previously set for the Agent:
   - `DD_OTLP_CONFIG_TRACES_SPAN_NAME_AS_RESOURCE_NAME`
   - `DD_OTLP_CONFIG_TRACES_SPAN_NAME_REMAPPINGS`

{{% /tab %}}
{{< /tabs >}}

## Enabling the new logic (opt-in)

> If you are using a version of the Datadog Agent or OpenTelemetry Collector prior to the versions listed in [Rollout Schedule](#default-rollout-schedule), you can enable it using the following methods. Datadog strongly recommends enabling this logic and adapting your monitors and dashboards.

Reviewer: ```suggestion
If you are using a version of the Datadog Agent or OpenTelemetry Collector prior to the versions listed in [Rollout Schedule](#default-rollout-schedule), you can enable the new logic using the following methods. Datadog strongly recommends enabling this logic and adapting your monitors and dashboards.
```

   ```

   **Note**: If you already have features configured with this variable, use a comma-separated list, ensuring you disable the correct flag:

   ```shell
   export DD_APM_FEATURES="existing_feature:true,disable_operation_and_resource_name_logic_v2"
   ```

{{% /tab %}}
{{% tab "DDOT Collector" %}}

Since DDOT enables this logic by default, you may need to disable it:

### Helm

Pass the feature gate flag to the embedded DDOT Collector using Helm values:

```shell
helm upgrade -i <RELEASE_NAME> datadog/datadog \
  -f datadog-values.yaml \
  --set datadog.otelCollector.featureGates="-datadog.EnableOperationAndResourceNameV2"
```

### Datadog Operator

If you are using Datadog Operator, use the `DD_APM_FEATURES` environment variable override in `datadog-agent.yaml`:

```yaml
# datadog-agent.yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  override:
    nodeAgent: # Or clusterAgent depending on where DDOT runs
      env:
        - name: DD_APM_FEATURES
          value: "disable_operation_and_resource_name_logic_v2" # Add comma-separated existing features if needed
```

{{% /tab %}}
{{< /tabs >}}

> ## Adapting Monitors, Dashboards, and custom configuration

Reviewer: ```suggestion
## Adapting monitors, dashboards, and custom configuration
```

After the new naming logic is active, take the following steps:

> 1.  **(Required) Update Monitors and Dashboards:**

Reviewer: ```suggestion
1.  **(Required) Update monitors and dashboards:**
```
